maintenance_plan_meter_mode/models/maintenance.py

```diff
@@ -0,0 +1,129 @@
+    meter_autoinc_value = fields.Float(
+        string='Value to self-increase per day')
+    meter_autoinc_unit = fields.Many2one(
+        'uom.uom',
+        string='Unit of Value to self-increase per day')
+    meter_autoinc_maxvalue = fields.Float(
+        string='Maximum amount since the last revision to generate another')
+    meter_autoinc_lastvalue = fields.Float(
+        string='Value from which the last revision was made')
+    meter_autoinc_lastupdate = fields.Date(
+        string='Date of last revision',
+        readonly=True)
+
+    @api.onchange('meter_mode')
+    def _onchange_meter_mode(self):
+        if self.meter_mode:
+            self.interval = None
+            self.interval_step = None
+            self.maintenance_plan_horizon = None
+            self.planning_step = None
+        else:
+            self.meter_current_value = None
+            self.meter_autoinc_value = None
+            self.meter_autoinc_unit = None
+            self.meter_autoinc_maxvalue = None
+            self.meter_autoinc_lastvalue = None
+            self.meter_autoinc_lastupdate = None
+
+    @api.constrains('meter_mode')
+    def _check_meter_mode(self):
+        for plan in self:
+            if plan.meter_mode:
+                if not plan.meter_autoinc_value or not plan.meter_autoinc_unit:
+                    raise ValidationError(
+                        _("It can´t be Self-increase mode"
+                          " without Value to self-increase per day"))
+                if not plan.meter_autoinc_maxvalue:
+                    raise ValidationError(
+                        _("It can´t be Self-increase mode"
+                          " without Maximum amount since the last revision"))
+
+    def _review_meter_mode_values(self, values):
+        meter_mode = values['meter_mode'] \
+            if 'meter_mode' in values else self.meter_mode
+        if meter_mode:
+            values['interval'] = None
+            values['interval_step'] = None
+            values['maintenance_plan_horizon'] = None
+            values['planning_step'] = None
+        else:
+            values['meter_current_value'] = None
+            values['meter_autoinc_value'] = None
+            values['meter_autoinc_unit'] = None
+            values['meter_autoinc_maxvalue'] = None
+            values['meter_autoinc_lastvalue'] = None
+            values['meter_autoinc_lastupdate'] = None
+        return values
+
+    @api.model
+    def create(self, values):
+        values = self._review_meter_mode_values(values)
+        return super(MaintenancePlan, self).create(values)
+
+    @api.multi
+    def write(self, values):
+        values = self._review_meter_mode_values(values)
+        return super(MaintenancePlan, self).write(values)
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = 'maintenance.equipment'
+
+    @api.model
+    def _cron_generate_requests(self):
+        # Generates maintenance request on the next_maintenance_date
```

Comment:
Missing `super()` call in order to generate regular plans requests:

```suggestion
    def _cron_generate_requests(self):
        super()._cron_generate_requests()
        # Generates maintenance request on the next_maintenance_date
```

Reply:
You're right. I forgot the super. I will do it.